src/app/georeferencer/qgsgeorefmainwindow.cpp

```diff
@@ -505,19 +505,27 @@ void QgsGeoreferencerMainWindow::addPoint( const QgsPointXY &pixelCoords, const
 
 void QgsGeoreferencerMainWindow::deleteDataPoint( QPoint coords )
 {
+  QgsGeorefDataPoint *pt = nullptr;
+  // This is the manhattan length from mouse cursor point to GCP
+  // We will use this distance to pick the closest GCP to the cursor
+  // GCPs farther than the initial minDistance are too far and are ignored
+  qreal minDistance = 10;
```

Comment:
instead of the hardcoded value -- could we use QgsMapTool::searchRadiusMU/QgsMapTool::searchRadiusMM() instead?

Reply:
I guess we could, but is there a real benefit from it?
AFAICT, `searchRadiusMM` is pretty much tied to the identify tool, so the options and code comment descriptions will have to be adapted too.
I can imagine someone raising this value to identify multiple features at once, or minimizing it for better identify precision, and then struggle with the georeferencer and his strange `searchRadiusMM` value choice!

Reply:
Hmm - I think we need an equivalent central method for map editing tools then. My concern with a hardcoded pixel value is that it's going to be completely inappropriate for certain setups (e.g. hidpi screens). We need something which accounts for dpi also, which is why a mm based size would be preferable.

Reply:
There is also the *Search radius for vertex edits* setting (`/qgis/digitizing/search_radius_vertex_edit`) that looks more appropriate, but it can also be set for map units which makes no sense for the georeferencer case...

Reply:
@nyalldawson I think the easiest solution is to just set the hardcoded value to a larger number if 10 is too small for hidpi displays (I cannot test).
A too small value will make the tool function like it does now, needing better aiming from the user.
A slightly large value will cause no real issue, since the GCP closest to the click point will be moved, the user will still expect this GCP to be moved.
A huge value could be a little strange as the closest GCP might be very far from the click position and the user might not make the connection.

Reply:
I think there's a better approach :)

We should:
1. Add a method in the QgsMapTool base class for "selection tolerance pixels" and use that here
2. The base class method would use a font metric based size (e.g. the width of a "-" character, so that it correctly returns a useful tolerance for hidpi setups

Reply:
I am not sure I completely understood your suggestion. The method should return 10 for low dpi and for example 40 on hidpi?
Given that I am not familiar with hidpi, how would the above work on a mixed dual screen setup? One window on a hidpi screen, one window on a low? Or even half a window on one screen and half on another?
I think I'll re-suggest to find a ballpark value good enough for both setups (I don't have a hidpi screen) and leave your approach for another PR.

Reply:
> Given that I am not familiar with hidpi, how would the above work on a mixed dual screen setup? One window on a hidpi screen, one window on a low? Or even half a window on one screen and half on another?

No, you're right -- it wouldn't. To handle this situation you'd need to add the "selectionTolerancePixels" api into QgsMapCanvas, because that will be aware of the screen that it's visible on.

> I think I'll re-suggest to find a ballpark value good enough for both setups (I don't have a hidpi screen) and leave your approach for another PR.

Unfortunately you won't find one -- it'll be either too big for use on lowdpi, or too small for usability on high dpi. The only approach is to take the dpi into account. I'm aware that many of our map tools are unusable on hidpi, which is why I'd like to see us start addressing this properly moving forward...

Reply:
> Unfortunately you won't find one -- it'll be either too big for use on lowdpi, or too small for usability on high dpi

If we had let's say a value of 10 effective hidpi pixels, what would that translate to normal pixels in lowdip displays? Even at 4 times the pixels, 40 is a perfectly usable number for lowdpi displays, way better than the one or two pixels that is the equivalent with the current functionality! At least for the georeferencer tools I don't think that too big can be an issue.

Reply:
```
>>> iface.mapCanvas().physicalDpiX()
210
```

Should be HiDPI

Reply:
Thanks @m-kuhn ,
my laptop's 1366x768 returns 118 and my desktop's 1920x1080 returns 72.

Would it make sense to just do `qreal minDistance = mCanvas->physicalDpiX() / 10;` ?

Reply:
That's what I'd propose.

Then, if possible, I would multiply with a conversion factor to convert inches to mm and make that configurable through the identify radius.

Reply:
The identify radius is pretty much tied to the identify tool and doesn't really make sense to be used here. If one wants a tighter or looser identify radius, he probably would not want that to affect vertex picking in the georeferencer.

Reply:
Possibly. But a common implementation for picking nodes for the node tool and gcp's here would make sense?

Reply:
Yes, that would make sense! But, the vertex tool uses for this purpose the setting `/qgis/digitizing/search_radius_vertex_edit`, which may be set in pixels but also in map units. And if set in map units it would not make sense for use with the georeferencer. That is why I think we should settle for a static value, maybe a percentage of the dpi, like in the previous comment.
Another approach would be to add a georeferencer only setting, but I know that adding new settings is frowned upon so I never proposed that.

Reply:
For search radius in pixels, we could convert this to reasonable values in mm `search_radius_map_tools` and reuse.

I can't actually think of a use case where the search radius in map units makes sense. Can you? If it's still required to have that for a good reason we could have two settings in parallel (one in mm, one in mu) -- or we just drop it.

Reply:
The issue is we now have `search_radius_vertex_edit` and `search_radius_vertex_edit_units` that defines if the former is in px or mu.

Truth is I cannot think of a use case for mu either, but it seems that the mu option preexisted and px was added later, plus I 'd hate to brake someones workflow just because I couldn't think of his use case!

Reply:
What about adding a new UnitType in QgsTolerance?

Reply:
I gave `qreal minDistance = mCanvas->physicalDpiX() / 10;` a shot and it works nice though slightly different on each of my two displays. Could you please test it on a hidpi setup?

Reply:
Gives around 7mm radius here

Reply:
I also get around 7mm on a 24" 1920x1080 display and around 3mm on a 13" 1366x768 display.
Both values are very usable, way better than the current 0.1mm imho! =p
What do you think?

Reply:
I think having involving dpi is a very good thing. And the number `10` should be configurable.

Reply:
Well, if it's configurable then there is no need for it to be dpi dependent, is it?